data/crac-io/crac-io-json/src/main/java/com/powsybl/openrao/data/craciojson/deserializers/InjectionSetpointArrayDeserializer.java

```diff
@@ -26,37 +28,88 @@ public final class InjectionSetpointArrayDeserializer {
     private InjectionSetpointArrayDeserializer() {
     }
 
-    public static void deserialize(JsonParser jsonParser, NetworkActionAdder ownerAdder, Map<String, String> networkElementsNamesPerId, String version) throws IOException {
+    private static void checkExpectedUnit(Unit expectedUnit, Unit unit, Identifiable<?> ne) {
+        if (!Objects.isNull(unit) && unit != expectedUnit) {
+            throw new OpenRaoException("Network element '" + ne.getId() + "' is a " + ne.getType() + ", its injection should be in " + expectedUnit.name() + " not in " + unit);
+        }
+    }
+
+    public static void deserialize(JsonParser jsonParser, NetworkActionAdder ownerAdder, Map<String, String> networkElementsNamesPerId, Network network) throws IOException {
         if (networkElementsNamesPerId == null) {
             throw new OpenRaoException(String.format("Cannot deserialize %s before %s", INJECTION_SETPOINTS, NETWORK_ELEMENTS_NAME_PER_ID));
         }
         while (jsonParser.nextToken() != JsonToken.END_ARRAY) {
-            InjectionSetpointAdder adder = ownerAdder.newInjectionSetPoint();
+            String networkElementId = null;
+            Integer setpoint = null;
+            Unit unit = null;
             while (!jsonParser.nextToken().isStructEnd()) {
                 switch (jsonParser.getCurrentName()) {
                     case NETWORK_ELEMENT_ID:
-                        String networkElementId = jsonParser.nextTextValue();
-                        if (networkElementsNamesPerId.containsKey(networkElementId)) {
-                            adder.withNetworkElement(networkElementId, networkElementsNamesPerId.get(networkElementId));
-                        } else {
-                            adder.withNetworkElement(networkElementId);
-                        }
+                        networkElementId = jsonParser.nextTextValue();
                         break;
                     case SETPOINT:
                         jsonParser.nextToken();
-                        adder.withSetpoint(jsonParser.getIntValue());
+                        setpoint = jsonParser.getIntValue();
```

Comment:
Should be a double.

Reply:
For iso functionnality, it was read as int (see line above)...

Reply:
Yes I saw that but isn't it a bug ? We are setting a injection setpoint, all double value in json will be truncated ?

Reply:
It will be truncated that's why I did not change that, I wanted to avoid functional change as much as possible in case it breaks the tests (especially the ones not run by the CI). But I added the fix for that, we will see. (I can also do a separate PR for this bug if we exêct it to break the tests).

Reply:
It does make more sense with a double indeed. If a test breaks because of it, then we will fix the test.